cpp/src/arrow/compute/exec/hash_join_benchmark.cc

```diff
@@ -132,7 +134,7 @@ class JoinBenchmark {
                                 left_keys, *r_batches_with_schema.schema, right_keys,
                                 filter, "l_", "r_"));
 
-    join_ = *HashJoinImpl::MakeBasic();
```

Comment:
I added a parameter to BenchmarkSettings, but I don't run both variants. It would be taking twice as much time and it would probably break graph producing script that @save-buffer wrote for post-processing hash join benchmark results, that may expect a single set of outputs.